### PR TITLE
feat(listing): API Listing

### PR DESCRIPTION
**Pull request by flesher1**
Adds default icon and banner images (Front end only) for the API listing
Adds the createEdit functionality
Removes screenshots from API listing page

Closes #AMLOS-766, 698

**Description**
An API listing with look slightly different on AML.  The create/edit page needs an update so that when a user selects to create an API some fields are hidden (graphics), and new fields are shown (schema/etc)

**Acceptance Criteria**:
A user can create an API listing from the Create/Edit page.

There may need to be further improvements made to listing view so that user can better understand API listings.

**how to test**
pull this branch and https://github.com/aml-development/ozp-backend/pull/450
you should see 2 new listings, both with the API banner.
You should be able to go into create edit and make a new API listing. You should not be required to add graphics.

### CODE REVIEW

File: src/components/createEdit/CreateEdit.js [hunk 1]
```diff
@@ -145,6 +152,14 @@ class CreateEdit extends React.Component {
         this.saveListing("PENDING")
     }
 
+    checkType(type){
+        if(type==="API"){
```
Comment (irtiza-rizvi) [inline src/components/createEdit/CreateEdit.js]: Should use `StaticListingType.api` here

**Comment by irtiza-rizvi** [comment]
Works as expected! :+1:

**Comment by flesher1** [comment]
This is finally done and ready.

**Comment by irtiza-rizvi** [comment]
@flesher1 I still give it a :+1:.  One suggestion I would have is to localize the code for getting the listing images so that API listing type conditionals aren't littered throughout the codebase.  A few approaches:

1. getter on the serializer `Listing` class:
```
export class Listing {
    // bunch of stuff
    get bannerImage() {
        return this.listingType === StaticListingType.api ? apiBanner : this.bannerIcon ? this.bannerIcon.url : '';
    }
```
Usage: ``

I'd also recommend moving `StaticListingType` to the serializer file and making `api` upper-case

2. Make a `helper.js` file and put functions in there to get the appropriate image based on type

3. Create a `ListingBannerImage` component:
```
class ListingBannerImage extends React.Component {
    render() {
        var { listing } = this.props;
        
        return (
            <img src={listing.listingType === StaticListingType.api ? apiBanner : listing.bannerIcon ? listing.bannerIcon.url : ''}/>
        )
    }
}
```
Usage: `<ListingBannerImage listing={listing}/>`

You can get fancy and try to make each approach re-usable for different image types (small icon, large icon, etc) or just create different functions/components for each.

**Comment by flesher1** [comment]
@irtiza-rizvi I will take that into consideration. It will likely end up being a function where i pass the listing object in and decide on the image to return. There are too many little differences between the location of the images to make good use out of a serializer centric method i think.

Will make an update at some point in the future to reflect the changes. I am also betting there will be changes required for the FE to further improve the differences between an API listing and a regular listing.

**Comment by blynch11p** [comment]
Search results when filtering on API listing type have the wrong icon.  Is this just a data issue?

**Comment by blynch11p** [comment]
Fix shows API images in search :+1:

**Pull request merged**